src/broker/Background.js

```diff
@@ -152,8 +152,16 @@ class Background {
   onExternalMessage (connection, { id, type, data }) {
     const { url } = connection.sender
     const { origin } = new URL(url)
-
-    const allowed = this.externalConnectionApprovalMap[origin]
+    let chain
+    if (data.chain) {
+      chain = data.chain
+    } else {
+      const { asset } = data
+      chain = assets[asset].chain
+    }
+    const { externalConnections, activeWalletId } = this.store.state
+    const allowed = Object.keys(externalConnections[activeWalletId] || {}).includes(origin) &&
```

Comment:
Shouldn't it be `|| []` ? I think object does not have includes

Reply:
got it I will make the same inside OneInchSwapProvider, about the includes it's because Im using Object.keys that returns an array

Reply:
@monokh I can't found the `OneInchSwapProvider ` in this branch also in dev